### PR TITLE
Port to 23w51b

### PR DESCRIPTION
**Pull request by modmuss50**
Breaking changes:

- `FabricBrewingRecipeRegistry.registerPotionRecipe` takes `RegistryEntry<Potion>` instead of `Potion`
- `SculkSensorFrequencyRegistry.regster` takes `RegistryKey<GameEvent>` instead of `GameEvent`
- `FabricLanguageProvider.add` takes `RegistryEntry<EntityAttribute>` instead of `EntityAttribute`
- `FabricTagProvider.GameEventTagProvider` was removed replace with `FabricTagProvider<GameEvent>`
- `FabricItem.getAttributeModifiers` returns a Multimap with a key of `RegistryEntry<EntityAttribute>` instead of `EntityAttribute`
- `ModifyItemAttributeModifiersCallback.modifyAttributeModifiers` takes Multimap with a key of `RegistryEntry<EntityAttribute>` instead of `EntityAttribute`

### CODE REVIEW

File: fabric-data-generation-api-v1/src/testmod/java/net/fabricmc/fabric/test/datagen/DataGeneratorTestEntrypoint.java [hunk 1]
```diff
@@ -238,7 +236,7 @@ public void generateTranslations(TranslationBuilder translationBuilder) {
 			translationBuilder.add(SIMPLE_BLOCK, "Simple Block");
 			translationBuilder.add(new Identifier(MOD_ID, "identifier_test"), "Identifier Test");
 			translationBuilder.add(EntityType.ALLAY, "Allay");
-			translationBuilder.add(EntityAttributes.GENERIC_ARMOR, "Generic Armor");
+			translationBuilder.add(EntityAttributes.GENERIC_ARMOR.value(), "Generic Armor");
```
Comment (maityyy) [inline fabric-data-generation-api-v1/src/testmod/java/net/fabricmc/fabric/test/datagen/DataGeneratorTestEntrypoint.java]: I think we need to add this, like we did with `CreativeModeTab`
```java
default void add(RegistryEntry<EntityAttribute> key, String translation) {
    add(key.value(), translation);
}
```
And then no breaking change
```java
translationBuilder.add(EntityAttributes.GENERIC_ARMOR, "Generic Armor");
```

Comment (modmuss50) [reply fabric-data-generation-api-v1/src/testmod/java/net/fabricmc/fabric/test/datagen/DataGeneratorTestEntrypoint.java]: Ah, yes good spot. Thanks

Comment (maityyy) [reply fabric-data-generation-api-v1/src/testmod/java/net/fabricmc/fabric/test/datagen/DataGeneratorTestEntrypoint.java]: Maybe also remove `add(EntityAttribute, String)` and leave only `add(RegistryEntry<EntityAttribute> key, String translation)`

Comment (modmuss50) [reply fabric-data-generation-api-v1/src/testmod/java/net/fabricmc/fabric/test/datagen/DataGeneratorTestEntrypoint.java]: Yep, thats what ive done 👍

File: fabric-transitive-access-wideners-v1/template.accesswidener [hunk 2]
```diff
@@ -44,7 +44,7 @@ transitive-accessible method net/minecraft/item/ItemPlacementContext <init> (Lne
 # Registering custom brewing recipes
 transitive-accessible method net/minecraft/recipe/BrewingRecipeRegistry registerItemRecipe (Lnet/minecraft/item/Item;Lnet/minecraft/item/Item;Lnet/minecraft/item/Item;)V
 transitive-accessible method net/minecraft/recipe/BrewingRecipeRegistry registerPotionType (Lnet/minecraft/item/Item;)V
-transitive-accessible method net/minecraft/recipe/BrewingRecipeRegistry registerPotionRecipe (Lnet/minecraft/potion/Potion;Lnet/minecraft/item/Item;Lnet/minecraft/potion/Potion;)V
+transitive-accessible method net/minecraft/recipe/BrewingRecipeRegistry registerPotionRecipe (Lnet/minecraft/registry/entry/RegistryEntry;Lnet/minecraft/item/Item;Lnet/minecraft/registry/entry/RegistryEntry;)V
```
Comment (maityyy) [inline fabric-transitive-access-wideners-v1/template.accesswidener:47]: AW still doesn't print an error or log if a class/field/method was not found after game update?

Comment (modmuss50) [reply fabric-transitive-access-wideners-v1/template.accesswidener:47]: It does, this caused the build to fail. See my 2nd commit fixing it.

**Pull request merged**